staging/src/k8s.io/client-go/tools/cache/store.go

```diff
@@ -82,6 +82,10 @@ type KeyError struct {
 
 // Error gives a human-readable description of the error.
 func (k KeyError) Error() string {
+	if err, ok := k.Err.(KeyError); ok {
```

Comment:
it's expected to have nested keyError in some cases, no need to flatten it, IMO. also object in the inner keyError should be the same as the object in the outer layer to avoid losing info.

Reply:
we need to use a `reflect.DeepEqual` to determine if two objects are the same, which I feel is a bit overloaded.

I am going to close this pr